lib/actions/user.js

```diff
@@ -181,3 +188,91 @@ export function deleteUserMonitoredTrip (id) {
     }
   }
 }
+
+/**
+ * Requests a verification code via SMS for the logged-in user.
+ */
+export function requestPhoneVerificationSms (originalUserData, newPhoneNumber) {
+  return async function (dispatch, getState) {
+    const { otp, user } = getState()
+    const { otp_middleware: otpMiddleware = null } = otp.config.persistence
+
+    if (otpMiddleware) {
+      const { accessToken } = user
+
+      // FIXME: (required by middleware) Only temporarily update the user's phone number
+      //   in the database with the pending one,
+      //   and cache the previous value and verification state so they can be reverted.
+      //   The full user record will be updated upon the user clicking Finish/Save preferences.
+      // TODO: Figure out what should happen if the user refreshes the browser at this stage.
+
+      // Make a clone of the original userData object.
+      const newUserData = clone(originalUserData)
+      const previousPhoneNumber = newUserData.phoneNumber
+      const previousIsPhoneNumberVerified = newUserData.isPhoneNumberVerified
+
+      newUserData.phoneNumber = newPhoneNumber
+      newUserData.isPhoneNumberVerified = false
+
+      const userUpdateResult = await updateUser(otpMiddleware, accessToken, newUserData)
+
+      if (userUpdateResult.status === 'success' && userUpdateResult.data) {
+        const sendSmsResult = await sendPhoneVerificationSms(otpMiddleware, accessToken, newUserData.id)
+        if (sendSmsResult.status === 'success') {
+          // Update application state on success.
+          dispatch(setCurrentUser({ accessToken, user: userUpdateResult.data }))
+        } else {
+          alert(`An error was encountered:\n${JSON.stringify(sendSmsResult)}`)
+
+          // FIXME: Also if there was an error in sending the verificaton request,
+          // revert the phone number and verification status in the database.
+          newUserData.phoneNumber = previousPhoneNumber
+          newUserData.isPhoneNumberVerified = previousIsPhoneNumberVerified
+          await updateUser(otpMiddleware, accessToken, newUserData)
+        }
+      } else {
+        alert(`An error was encountered:\n${JSON.stringify(userUpdateResult)}`)
+      }
+    }
+  }
+}
+
+/**
+ * Validate the phone number verification code for the logged-in user.
+ */
+export function verifyPhoneNumber (originalUserData, code) {
+  return async function (dispatch, getState) {
+    const { otp, user } = getState()
+    const { otp_middleware: otpMiddleware = null } = otp.config.persistence
+
+    if (otpMiddleware) {
+      const { accessToken } = user
+      const sendResult = await validatePhoneVerificationCode(otpMiddleware, accessToken, originalUserData.id, code)
```

Comment:
This util function is used once. I don't really see the value in having the fetch code isolated, especially with a method that has four args.

Reply:
Will address once https://github.com/opentripplanner/otp-react-redux/pull/246 is merged.